File: Selenium/Activities/Activity22.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity22 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        Actions builder = new Actions(driver);

        // Open a page
        driver.get("https://www.training-support.net/selenium/popups");

        // Page title
        System.out.println("Title of page is: " + driver.getTitle());

        //Sign in button
        WebElement signIn = driver.findElement(By.cssSelector(".inverted"));
        builder.moveToElement(signIn).build().perform();

        String msg = signIn.getAttribute("data-tooltip");
        System.out.println(msg);
        signIn.click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("username")));

        // Enter the credentials
        driver.findElement(By.id("username")).sendKeys("admin");
        driver.findElement(By.id("password")).sendKeys("password");
        Thread.sleep(1000);
        driver.findElement(By.xpath("//button[text()='Log in']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("action-confirmation")));
        WebElement loginMsg = driver.findElement(By.id("action-confirmation"));
        System.out.println("Login Message: " + loginMsg.getText());

        driver.quit();

    }
}

File: Selenium/Activities/Activity22.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the link in browser
    driver.get("https://www.training-support.net/selenium/popups")

    wait = WebDriverWait(driver, 10)

    # Print page title
    print("Page Title is: " + driver.title)

    # Sign in button
    signIn = driver.find_element(By.CSS_SELECTOR, ".inverted")
    msg = signIn.get_attribute("data-tooltip")
    print(msg)
    signIn.click()

    # Enter the credentials
    driver.find_element(By.ID, "username").send_keys("admin")
    driver.find_element(By.ID, "password").send_keys("password")
    driver.implicitly_wait(20)
    driver.find_element(By.XPATH, "//button[text()='Log in']").click()

    loginMsg = driver.find_element(By.ID, "action-confirmation")
    print("Login Message: " + loginMsg.text)

    driver.quit()